src/sparse_ir/sve.py

```diff
@@ -30,41 +30,57 @@ def compute(K, eps=None, n_sv=None, n_gauss=None, dtype=float, work_dtype=None,
     using a collocation).
 
     Arguments:
-
-      - ``K``: Integral kernel to take SVE from
-      - ``eps``:  Relative cutoff for the singular values. Defaults to double
-        precision (2.2e-16) if the xprec package is available, and 1.5e-8
-        otherwise.
-      - ``n_sv``: Maximum basis size. If given, only at most the ``n_sv`` most
-        significant singular values and associated singular functions are
-        returned.
-      - ``n_gauss``: Order of Legendre polynomials. Defaults to hinted value
-        by the kernel.
-      - ``dtype``: Data type of the result.
-      - ``work_dtype``: Working data type. Defaults to a data type with
-        machine epsilon of at least ``eps**2``, or otherwise most accurate data
-        type available.
-      - ``sve_strat``: SVE to SVD translation strategy. Defaults to SamplingSVE.
-      - ``svd_strat``: SVD solver. Defaults to fast (ID/RRQR) based solution
-         when accuracy goals are moderate, and more accurate Jacobi-based
-         algorithm otherwise.
-
-    Return tuple ``(u, s, v)``, where:
-
-     - ``u`` is a ``PiecewiseLegendrePoly`` instance holding the left singular functions
-     - ``s`` is a vector of singular values
-     - ``v`` is a ``PiecewiseLegendrePoly`` instance holding the right singular functions
+        K (kernel.AbstractKernel):
+            Integral kernel to take SVE from
+        eps (float):
+            Accuracy target for the basis: attempt to have singular values down
+            to a relative magnitude of ``eps``, and have each singular value
+            and singular vector be accurate to ``eps``.  A ``work_dtype`` with
+            a machine epsilon of ``eps**2`` or lower is required to satisfy
+            this. Defaults to ``2.2e-16`` if xprec is available, and ``1e-8``
+            otherwise.
+        cutoff (float):
+            Relative cutoff for the singular values.  A ``work_dtype`` with
+            machine epsilon of ``cutoff`` is required to satisfy this.
+            Defaults to a small multiple of the machine epsilon.
+
+            Note that ``cutoff`` and ``eps`` serve distinct purposes. ``cutoff``
+            reprsents the accuracy to which the kernel is reproduced, whereas
+            ``eps`` is the accuracy to which the singular values and vectors
+            are guaranteed.
```

Comment:
I got confused. In `__init__` of `FiniteTempBasis`, `eps` seems to mean a cutoff for singular values. Is it correct?

Reply:
Yes, but that happens only at the level of constructing the basis.

The problem I am trying to solve is the following: suppose you want to construct sampling points for a basis of size L the "polynomial way", i.e., by using the roots of the L'th basis function (rather than the way we do it now by using the extrema of the highest-order basis function plus some points we guess).  This however means you need *more* basis function than given by a cutoff.  However, you don't really care about the *relative* accuracy of these functions, as long as they have the roots in roughly the right place.

One way to do this is to have two truncation parameters for your SVD/SVE: one, include as many singular values so as to represent the kernel with a given accuracy, i.e., `cutoff`, and two, include make sure that all singular values above a level of `eps` have accuracy `eps`.  One uses only those basis functions which match the first criteria in constructing the basis, but the  "extra" basis functions (those which match the first criterion but not the second) can be used to construct sampling points (think of augmented basis etc.)  This is not implemented yet.